HotelReservation/HotelReservation/Hotel/HotelSystemReservationSteps/HotelSystemReservationStepCreator.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace HotelReservation.Hotel.HotelSystemReservationSteps
```

Comment:
Po co taka generalizacja kroków? Czy flow konsoli się jakkolwiek zmienia? Czy jest konfigurowalny? Pocięcie większej logiki na mniejszcze części, wydzielenie metod czy klas w celu zwiększenia czytelności jest jak najbardziej dobrą praktyką. Tworzenie dodatkowej logiki która dostarcza koszty utrzymania i zrozumienia już nie.

StepExecutor jest odpowiedzialny za odpalanie:
* Kroków rezerwacji hotelu
* Kroków menu konsoli

To są absolutnie dwa różne koncepty, które są na siłę (IConvertible) obsłużone przez jedną klasę.

Reply:
Fixed - kroki obsługujące konsolę nie są już odpalane przez stepsExecutor. SelectHotel() na pewno do poprawy - być może powinna zwracać obiekt Hotel, a nie ID.